apps/backend/src/groups/groups.service.ts

```diff
@@ -142,4 +171,46 @@ export class GroupsService {
 
     return groupToDelete;
   }
+
+  // This method ensures that the passed in user is in all of the
+  // passed in groups, as long as the group already exists.
+  // It will additionally remove the user from any groups not in the list.
+  // Called from oidc.strategy.ts, if OIDC_EXTERNAL_GROUPS is enabled
+  async syncUserGroups(user: User, groups: string[]) {
+    const currentGroups = await user.$get('groups', {include: [User]});
+    const groupsToLeave = currentGroups.filter(
+      (group) => !groups.includes(group.name)
+    );
+    const existingGroups: Group[] = [];
```

Comment:
should be instantiated closer to where it's being used (the find existing groups block)